models/intermediate/int_zendesk__schedule_spine.sql

```diff
@@ -0,0 +1,106 @@
+{{ config(enabled=var('using_schedules', True)) }}
+
+with timezone as (
+
+    select *
+    from {{ var('time_zone') }}
+
+), daylight_time as (
+
+    select *
+    from {{ var('daylight_time') }}
+
+), schedule as (
+
+    select *
+    from {{ var('schedule') }}   
+
+), timezone_with_dt as (
+
+    select 
+        timezone.*,
+        daylight_time.daylight_start_utc,
+        daylight_time.daylight_end_utc,
+        daylight_time.daylight_offset_minutes
+
+    from timezone 
+    left join daylight_time 
+        on timezone.time_zone = daylight_time.time_zone
+
+), order_timezone_dt as (
+
+    select 
+        *,
+        -- will be null for timezones without any daylight savings records (and the first entry)
+        -- we will coalesce the first entry date with .... the X years ago
+        lag(daylight_end_utc, 1) over (partition by time_zone order by daylight_end_utc asc) as last_daylight_end_utc,
+        -- will be null for timezones without any daylight savings records (and the last entry)
+        -- we will coalesce the last entry date with the current date 
+        lead(daylight_start_utc, 1) over (partition by time_zone order by daylight_start_utc asc) as next_daylight_start_utc
+
+    from timezone_with_dt
+
+), split_timezones as (
+
+    -- standard schedule (includes timezones without DT)
+    -- starts: when the last Daylight Savings ended
+    -- ends: when the next Daylight Savings starts
+    select 
+        time_zone,
+        standard_offset_minutes as offset_minutes,
+
+        -- last_daylight_end_utc is null for the first record of the time_zone's daylight time, or if the TZ doesn't use DT
+        coalesce(last_daylight_end_utc, cast('1970-01-01' as date)) as valid_from,
```

Comment:
I noticed that for any record that isn't the first, the `valid_from` and previous `valid_until` are the exact same timestamp. Could this possibly cause issues when doing joins downstream? I wonder if we would need the `valid_from` to be like a second after the previous `valid_until` so there is no overlap. Thoughts?